Module#04/security-service/src/main/java/com/epam/esm/security/service/AuthenticationService.java

```diff
@@ -0,0 +1,235 @@
+            final HttpServletResponse response) {
+        String authHeader = request.getHeader("Authorization");
+        if (authHeader != null && authHeader.startsWith("Bearer ")) {
+            String jwt = authHeader.substring(7);
+            Token token = provider.findByToken(jwt).orElse(null);
+            if (token != null) {
+                token.setExpired(true);
+                token.setRevoked(true);
+                provider.save(token);
+            }
+        }
+        response.setStatus(HttpServletResponse.SC_OK);
+        try {
+            response.getWriter().write("Logout successful");
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    /**
+     * Sets the authentication token for the provided username and password.
+     *
+     * @param request The AuthenticationRequest containing the username and password.
+     */
+    private void setAuthenticationToken(
+            final AuthenticationRequest request) {
+        Authentication authenticate = manager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        request.getUsername(),
+                        request.getPassword()));
+        SecurityContextHolder.getContext()
+                .setAuthentication(authenticate);
+    }
+
+    /**
+     * Retrieves the user with the given username.
+     *
+     * @param username The username of the user to find.
+     * @return The User object.
+     * @throws UserNotFoundException if the user is not found.
+     */
+    @Transactional
+    public User findUser(final String username) {
+        return userRepository.findByUsername(username)
+                .orElseThrow(() -> new UserNotFoundException(
+                        format("User not found: %s", username)));
+    }
+
+    /**
+     * Saves a new user with the provided registration details and default role.
+     *
+     * @param request The RegisterRequest containing the user registration details.
+     * @return The saved User object.
+     */
+    @Transactional
+    public User saveUserWithRole(
+            final RegisterRequest request) {
+        User user = User.builder()
+                .password(encoder.encode(request.getPassword()))
+                .username(request.getUsername())
+                .email(request.getEmail())
+                .role(getRole())
+                .build();
+        return userRepository.save(user);
+    }
+
+    /**
+     * Finds the user with the given username.
+     *
+     * @param username The username of the user to find.
+     * @return Optional containing the User object, or empty if not found.
+     */
+    @Transactional
+    public Optional<User> findByUsername(String username) {
+        return userRepository.findByUsername(username);
+    }
+
+    /**
+     * Returns the Role object representing the default role for a user.
+     *
+     * @return The default Role object.
+     */
+    private static Role getRole() {
```

Comment:
it must be renamed to getUserRole or any other, it is not clear which role is User, Admin...

Reply:
> it must be renamed to getUserRole or any other, it is not clear which role is User, Admin...

> Renamed the method to getDefaultUserRole(), making it clearer that it returns the default role for the user